Snakefile

```diff
@@ -0,0 +1,197 @@
+import pandas as pd
+
+genome_acc_df = pd.read_csv("inputs/genome_contaminants.csv") 
+GENOME_ACC = genome_acc_df['ident'].unique().tolist()
+sra_acc_df = pd.read_csv("inputs/seq_contaminants.csv")
+SRA_ACC = sra_acc_df['ident'].unique().tolist()
+KSIZE = [21]
+
+rule all:
+    input: 
+        expand("outputs/sourmash_contam_db/contamdb.dna.k{ksize}.zip", ksize = KSIZE),
+        "outputs/sourmash_contam_db/contamdb.taxonomy.csv"
+
+#############################################
+## SUBSET GTDB REPS DATABASE
+## covers:
+## - kit contamination
+## - human microbiome contamination
+#############################################
+
+rule download_gtdb_rs207_metadata:
+    output: "inputs/gtdb/bac120_metadata_r207.tar.gz"
+    shell:'''
+    curl -JLo {output} https://data.gtdb.ecogenomic.org/releases/release207/207.0/bac120_metadata_r207.tar.gz
+    '''
+
+rule decompress_gtdb_rs207_metadata:
+    input: "inputs/gtdb/bac120_metadata_r207.tar.gz"
+    output: "inputs/gtdb/bac120_metadata_r207.tsv"
+    params: outdir = "inputs/gtdb/"
+    shell:'''
+    tar xf {input} -C {params.outdir}
+    '''
+
+rule download_gtdb_rs207_taxonomy:
+    output: "inputs/sourmash_databases/gtdb-rs207.taxonomy.csv.gz"
+    shell:'''
+    curl -JLo {output} https://osf.io/v3zmg/download
+    '''
+
+rule download_sourmash_db:
+    input: "inputs/sourmash_databases/sourmash-database-info.csv"
+    output: "inputs/sourmash_databases/gtdb-rs207.genomic-reps.dna.k{ksize}.zip"
+    run:
+        sourmash_database_info = pd.read_csv(str(input[0]))
+        ksize = int(wildcards.ksize)
+        db_df = sourmash_database_info.loc[(sourmash_database_info['ksize'] == ksize)]
+        osf_hash = db_df['osf_hash'].values[0] 
+        shell("curl -JLo {output} https://osf.io/{osf_hash}/download")
+
+
+rule sourmash_sig_describe_db:
+    '''
+    produce a csv file that describes each of the signatures in the database.
+    this CSV file will be post-processed to use as a picklist to extract signatures of interest form the sourmash db
+    '''
+    input: "inputs/sourmash_databases/gtdb-rs207.genomic-reps.dna.k{ksize}.zip"
+    output: "outputs/sourmash_sig_describe/gtdb-rs207.genomic-reps.dna.k{ksize}.csv"
+    conda: "envs/sourmash.yml"
+    shell:'''
+    sourmash sig describe -k {wildcards.ksize} --csv {output} {input}
+    '''
+
+rule subset_sourmash_db:
+    input: 
+        sig_describe = "outputs/sourmash_sig_describe/gtdb-rs207.genomic-reps.dna.k{ksize}.csv",
+        contams = "inputs/doi10.1016j.tim.2018.11.003-table1.csv",
+        gtdb_metadata = "inputs/gtdb/bac120_metadata_r207.tsv"
+    output: picklist="outputs/sourmash_sig_describe_subset/picklist_k{ksize}.csv"
+    conda: "envs/tidyverse.yml"
```

Comment:
oh I see this is where the tidyverse environment is called, it's defintely a snakemake thing I missed before